otter/integration/lib/nova.py

```diff
@@ -83,3 +90,49 @@ def delete_servers(server_ids, rcs, pool=None, _treq=treq):
     """
     return gatherResults([NovaServer(id=_id, pool=pool, treq=_treq).delete(rcs)
                           for _id in server_ids])
+
+
+def list_servers(rcs, pool, _treq=treq):
+    """
+    Get a list of all servers, with an optional name regex provided.  This
+    does not handle pagination, and instead just increases the limit to an
+    absurdly high number.
+    """
+    params = {'limit': 10000}
+    return _treq.get(
+        "{}/servers/detail".format(rcs.endpoints['nova']),
+        params=params,
+        headers=headers(str(rcs.token)),
+        pool=pool
+    ).addCallback(check_success, [200]).addCallback(_treq.json_content)
+
+
+def wait_for_servers(rcs, pool, group, matcher, timeout=600, period=10,
+                     clock=None, _treq=treq):
+    """
+    Wait until Nova reaches a particular state with regards to the servers for
+    the given group.
```

Comment:
Usually I don't ask for much from docstrings in test code, but I think this could use a bit more information about what "a particular state" means -- i.e. documentation that `matcher` is a testtools matcher.